### PR TITLE
add JSON_ARRAY() function

### PR DESCRIPTION
**Pull request by jennifersp**

### CODE REVIEW

File: enginetest/queries.go [hunk 1]
```diff
@@ -615,6 +615,27 @@ var QueryTests = []QueryTest{
...
+		},
+	},
+	{
+		Query: `SELECT JSON_ARRAY('a', 1, datetime('2020-01-01T12:00:00'))`,
+		Expected: []sql.Row{{
+			sql.MustJSON(`["a", 1, "2020-01-01 12:00:00 +0000 UTC"]`)},
+		},
+	},
+	{
+		Query: `SELECT JSON_ARRAY(pk, c1, c2, c3) FROM jsontable`,
```
Comment (VinaiRachakonda) [inline enginetest/queries.go:631]: Add test for empty set. Add additional tests that compose JSON_ARRAY with additional JSON functions to ensure type validity

Comment (VinaiRachakonda) [reply enginetest/queries.go:631]: You should also exercise this function outside of a table.

File: sql/expression/function/json_array.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+)
+
+// JSON_ARRAY([val[, val] ...])
+//
+// JSONArray Evaluates a (possibly empty) list of values and returns a JSON array containing those values.
+//
+// https://dev.mysql.com/doc/refman/8.0/en/json-creation-functions.html#function_json-array
+
+type JSONArray struct {
+	Docs []sql.Expression
```
Comment (VinaiRachakonda) [inline sql/expression/function/json_array.go:32]: 1. No reason to export this
2. `Docs` is not the right name. This function does not necessarily have to take in a json doc. Use `vals` or `args` or something more expressive.

File: sql/expression/function/json_array.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+// FunctionName implements sql.FunctionExpression
+func (j JSONArray) FunctionName() string {
+	return "json_array"
+}
+
+// Description implements sql.FunctionExpression
+func (j JSONArray) Description() string {
+	return "creates JSON array."
```
Comment (VinaiRachakonda) [inline sql/expression/function/json_array.go:49]: creates a JSON array

File: sql/expression/function/json_array.go [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+			jsonDoc, err = j.Type().Convert(jsonDoc)
+			if err != nil {
+				return nil, err
+			}
+
+			js := jsonDoc.(sql.JSONDocument).Val
+
+			switch js.(type) {
+			case time.Time:
+				t := js.(time.Time)
```
Comment (VinaiRachakonda) [inline sql/expression/function/json_array.go:120]: Curious about this. Does sql.JSONDocument not convert time to its string representation. Not sure why this is necessary.

File: sql/expression/function/json_array.go [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+	var resultArray = make([]interface{}, len(j.Docs))
+
+	for i, doc := range j.Docs {
+		jsonDoc, err := doc.Eval(ctx, row)
+		if err != nil {
+			return nil, err
+		}
+
+		if jsonStr, ok := jsonDoc.(string); ok {
+			resultArray[i] = jsonStr
```
Comment (VinaiRachakonda) [inline sql/expression/function/json_array.go:109]: You should run the string type through the conversion method. There might be some escaping properties that get missed.

**Review by VinaiRachakonda — APPROVED** [review]
LGTM!

**Review by VinaiRachakonda — APPROVED** [review]
